server/services/beatmaps.py

```diff
@@ -0,0 +1,269 @@
+from datetime import datetime
+from datetime import timedelta
+
+from server.adapters import osu_api_v2
+from server.errors import ServiceError
+from server.repositories import beatmaps
+from server.repositories.beatmaps import Beatmap
+from server.repositories.beatmaps import BeatmapRankedStatus
+
+
+def _create_beatmap_filename(
+    artist: str,
+    title: str,
+    version: str,
+    creator: str,
+) -> str:
+    return f"{artist} - {title} ({creator}) [{version}].osu"
+
+
+def _transform_to_internal_model(osu_api_beatmap: osu_api_v2.Beatmap) -> Beatmap:
+    assert osu_api_beatmap.beatmap_md5 is not None
+    assert osu_api_beatmap.last_updated is not None  # TODO?
+    assert osu_api_beatmap.beatmapset is not None
+
+    return {
+        "beatmap_id": osu_api_beatmap.beatmap_id,
+        "beatmap_set_id": osu_api_beatmap.beatmap_set_id,
+        "ranked_status": osu_api_beatmap.ranked_status,
+        "beatmap_md5": osu_api_beatmap.beatmap_md5,
+        "artist": osu_api_beatmap.beatmapset.artist,
+        "title": osu_api_beatmap.beatmapset.title,
+        "version": osu_api_beatmap.version,
+        "creator": osu_api_beatmap.beatmapset.creator_name,
+        "filename": _create_beatmap_filename(
+            artist=osu_api_beatmap.beatmapset.artist,
+            title=osu_api_beatmap.beatmapset.title,
+            version=osu_api_beatmap.version,
+            creator=osu_api_beatmap.beatmapset.creator_name,
+        ),
+        "total_length": osu_api_beatmap.total_length,
+        "max_combo": osu_api_beatmap.max_combo or 0,
+        "ranked_status_manually_changed": False,  # manually ranked
```

Comment:
if this is being tracked, may it be a good idea to track bancho's status vs ours?

Reply:
yeah you're right, also i think we want to do the same for updated_at

both bancho's and ours

Reply:
oh yeah definitely, i missed out that one